syncer/syncer.go

```diff
@@ -1706,6 +1711,10 @@ func (s *Syncer) handleQueryEvent(ev *replication.QueryEvent, ec eventContext) e
 			needHandleDDLs = appliedSQLs // maybe nil
 		}
 
+		if err := s.flushCheckPoints(); err != nil {
```

Comment:
If we do not dispatch a `flush` job, then we must call `s.jobWg.Wait()` before `s.flushCheckPoints()`.

(Is dispatch a `flush` job better?)